plugins/reference-assets/components/src/index.tsx

```diff
@@ -140,3 +166,49 @@ Info.Subtitle = SubtitleSlot;
 Info.PrimaryInfo = PrimaryInfoSlot;
 Info.Actions = ActionsSlot;
 Info.Footer = slotFactory("footer");
+
+export const Choice = (
+  props: Omit<AssetPropsWithChildren<ChoiceAsset>, "binding"> & {
+    /** The binding */
+    binding: BindingTemplateInstance;
+  },
+) => {
+  const { binding, children, ...rest } = props;
+  return (
+    <Asset type="choice" {...rest}>
+      <property name="binding">{binding.toValue()}</property>
+      {children}
+    </Asset>
+  );
+};
+
+Choice.Title = TitleSlot;
+Choice.Note = NoteSlot;
+Choice.Items = ItemsSlot;
+
+/**
+ * Comp for a choice item
+ */
+const ChoiceItem = (
+  props: WithChildren<
+    WithApplicability<WithTemplateTypes<Optional<ChoiceItemType, "id">>>
```

Comment:
Instead of defining and using the `Optional` type, I would use the same pattern we use with `Omit` to pull out the `binding` property on the Choice asset to just make `id` conditional